### PR TITLE
Cache IsMono call

### PR DESCRIPTION
**Pull request by cdmihai**
VS has potentially expensive TypeResolve events, so cache IsMono. Apparently the c++ side of things even triggers a design time build during some TypeResolve calls.

### CODE REVIEW

**Comment by cdmihai** [comment]
@radical

File: src/Shared/NativeMethodsShared.cs [hunk 1]
```diff
@@ -454,6 +454,12 @@ public SystemInformationData()
 
         #region Member data
 
+        static NativeMethodsShared()
```
Comment (rainersigwald) [inline src/Shared/NativeMethodsShared.cs]: Use a `Lazy<T>` instead of this static constructor?

Comment (cdmihai) [inline src/Shared/NativeMethodsShared.cs]: I'm not sure that helps, since a log of stuff calls into IsMono including anything that logs events.

Comment (rainersigwald) [inline src/Shared/NativeMethodsShared.cs]: Fair enough.

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Desktop please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Mono please

**Pull request merged**